server/service/labels.go

```diff
@@ -300,15 +301,20 @@ func listLabelsEndpoint(ctx context.Context, request interface{}, svc fleet.Serv
 	return resp, nil
 }
 
-func (svc *Service) ListLabels(ctx context.Context, opt fleet.ListOptions) ([]*fleet.Label, error) {
+func (svc *Service) ListLabels(ctx context.Context, opt fleet.ListOptions, includeHostCounts *bool) ([]*fleet.Label, error) {
 	if err := svc.authz.Authorize(ctx, &fleet.Label{}, fleet.ActionRead); err != nil {
 		return nil, err
 	}
-	vc, ok := viewer.FromContext(ctx)
-	if !ok {
-		return nil, fleet.ErrNoContext
+
+	filter := fleet.TeamFilter{}
+	// Default to including host counts.
+	if includeHostCounts == nil || *includeHostCounts {
+		vc, ok := viewer.FromContext(ctx)
+		if !ok {
+			return nil, fleet.ErrNoContext
+		}
```

Comment:
Let's pull this out of the if so we have less of a diverging code path here.

Also feels like the interface on the datastore method would be clearer if the param was a *User, with the param named `getHostCountsForTeamsVisibleToUser` or similar. Underlying behavior would still use the TeamFilter, but for the 3 places (including a test) where this is used that makes things a little clearer on what's going on.